hedera-node/src/main/java/com/hedera/services/ServicesState.java

```diff
@@ -159,143 +129,59 @@ public long getClassId() {
 
 	@Override
 	public int getVersion() {
-		return MERKLE_VERSION;
+		return StateVersions.CURRENT_VERSION;
 	}
 
 	@Override
 	public int getMinimumChildCount(int version) {
-		switch (version) {
-			case RELEASE_0160_VERSION:
-				return ChildIndices.NUM_0160_CHILDREN;
-			case RELEASE_0150_VERSION:
-				return ChildIndices.NUM_0150_CHILDREN;
-			case RELEASE_0140_VERSION:
-				return ChildIndices.NUM_0140_CHILDREN;
-			case RELEASE_0130_VERSION:
-				return ChildIndices.NUM_0130_CHILDREN;
-			case RELEASE_0120_VERSION:
-				return ChildIndices.NUM_0120_CHILDREN;
-			case RELEASE_0110_VERSION:
-				return ChildIndices.NUM_0110_CHILDREN;
-			case RELEASE_0100_VERSION:
-				return ChildIndices.NUM_0100_CHILDREN;
-			case RELEASE_090_VERSION:
-				return ChildIndices.NUM_090_CHILDREN;
-			case RELEASE_080_VERSION:
-				return ChildIndices.NUM_080_CHILDREN;
-			case RELEASE_070_VERSION:
-				return ChildIndices.NUM_070_CHILDREN;
-			default:
-				throw new IllegalArgumentException(String.format(UNSUPPORTED_VERSION_MSG_TPL, version));
+		if (version == StateVersions.RELEASE_0160_VERSION) {
+			return LegacyStateChildIndices.NUM_0160_CHILDREN;
+		} else if (version == StateVersions.RELEASE_0170_VERSION ) {
+			return StateChildIndices.NUM_0170_CHILDREN;
+		} else {
+			throw new IllegalArgumentException("Argument 'version='" + version + "' is invalid!");
```

Comment:
so we will only be backward compatible with one old version?

Reply:
Yep!